diffxpy/unit_test/test_continuous.py

```diff
@@ -0,0 +1,105 @@
+        """
+        Test if de.test.continuous() generates a uniform p-value distribution in the wald test
+        if it is given data simulated based on the null model. Returns the p-value
+        of the two-side Kolmgorov-Smirnov test for equality of the observed 
+        p-value distriubution and a uniform distribution.
+
+        :param n_cells: Number of cells to simulate (number of observations per test).
+        :param n_genes: Number of genes to simulate (number of tests).
+        """
+        logging.getLogger('diffxpy').addFilter('DEBUG')
+
+        sim = Simulator(num_observations=n_cells, num_features=n_genes)
+        sim.generate_sample_description(num_batches=0, num_conditions=0)
+        sim.generate()
+
+        random_sample_description = pd.DataFrame({
+            "pseudotime": np.random.random(size=sim.num_observations)
+        })
+
+        test = de.test.continuous_1d(
+            data=sim.X,
+            continuous="pseudotime",
+            df=3,
+            formula_loc="~ 1 + pseudotime",
+            formula_scale="~ 1",
+            factor_loc_totest="pseudotime",
+            test="wald",
+            sample_description=random_sample_description,
+            quick_scale=True,
+            batch_size=None
+        )
+        summary = test.summary()
+
+        # Compare p-value distribution under null model against uniform distribution.
+        pval_h0 = stats.kstest(test.pval, 'uniform').pvalue
+
+        print('KS-test pvalue for null model match of wald(): %f' % pval_h0)
+        print(test.pval)
+
+        assert pval_h0 > 0.05, "KS-Test failed: pval_h0 is <= 0.05!"
+
+        return pval_h0
+
+    def test_null_distribution_lrt(self, n_cells: int = 10, n_genes: int = 2):
+        """
+        Test if de.test.continuous() generates a uniform p-value distribution in lrt
+        if it is given data simulated based on the null model. Returns the p-value
+        of the two-side Kolmgorov-Smirnov test for equality of the observed
+        p-value distriubution and a uniform distribution.
+
+        :param n_cells: Number of cells to simulate (number of observations per test).
+        :param n_genes: Number of genes to simulate (number of tests).
+        """
+        logging.getLogger('diffxpy').addFilter('DEBUG')
+
+        sim = Simulator(num_observations=n_cells, num_features=n_genes)
+        sim.generate_sample_description(num_batches=0, num_conditions=0)
+        sim.generate()
+
+        random_sample_description = pd.DataFrame({
+            "pseudotime": np.random.random(size=sim.num_observations)
+        })
+
+        test = de.test.continuous_1d(
+            data=sim.X,
+            continuous="pseudotime",
+            df=3,
+            formula_loc="~ 1 + pseudotime",
+            formula_scale="~ 1",
+            factor_loc_totest="pseudotime",
+            test="lrt",
+            sample_description=random_sample_description,
+            quick_scale=True,
+            batch_size=None
```

Comment:
You should use `dtype="float64"`.
I spend two weeks on resolving numerical instabilities of the LRT null model test, only to find out that usage of "float32" is numerically too imprecise for this.

Reply:
I also changed the other null model tests to float64, just to be sure.

Reply:
Ok, I did this explicitly but this was happening per default already https://github.com/theislab/diffxpy/blob/c37eee88187b3b850d188e4958cc34f701507fa4/diffxpy/testing/base.py#L3215. Done in https://github.com/theislab/diffxpy/commit/d33853cccd3244267203ce5c47f45699fe27807d.